rpm/gen-fixtures-delta.sh

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env sh
+#
+# Generate RPM repository with DRPM fixture data. 
+# Several RPM utilities must be installed, including `createrepo` and `makedeltarpm`.
+#
+set -euo pipefail
+
+# Print a message to stdout explaining how to use this script.
+#
+# NOTE: $0 corresponds to the the script's name only in some shells and in some
+# contexts.
+show_help() {
+fmt <<EOF
+usage: gen-fixtures-delta.sh <output_dir> <assets_dir> <dpkg_name>
+
+Create <output_dir>. Use packages in <assets_dir> to generate DRPM fixture data 
+for package <dpkg_name> and place the results into <output_dir>. 
+A suggested name for <output_dir> is "zoo".
+
+EOF
+cat <<EOF
+<output_dir>
+    The directory into which generated fixtures are placed.
+<assets_dir>
+    The directory from which source material is read.
+ <dpkg_name>
+    The name of package which delta RPM will be generated for
+EOF
+}
+
+# Given a path to an RPM package, echo the package name.
+#
+# e.g. assets/walrus-baby-5.21-1.noarch.rpm → walrus-baby
+get_rpm_name() {
+    local filename="$(basename "${1}")"
+    IFS=$'\n' local parts=($(echo "${filename}" | tr - "\n"))
+    local end=$(expr ${#parts[@]} - 2)
+    for ((i=0; i<${end}; i++)); do
+        rpm_name+="${parts[i]}"-
+    done
+    rpm_name="${rpm_name::-1}"
+    echo "${rpm_name#*\!}"  # strip epoch
+}
+
+# Given a path to an RPM package, echo the package version.
+#
+# e.g. assets/walrus-5.21-1.noarch.rpm → 5.21
+get_rpm_version() {
+    local filename="$(basename "${1}")"
+    IFS=$'\n' local parts=($(echo "${filename}" | tr - "\n"))
+    echo "${parts[-2]}"
+}
+
+# Given a pathto an RPM package, echo the package release.
+#
+# e.g. assets/yum-cron-3.4.3-10.fc16.noarch.rpm  → 10.fc16
+get_rpm_release() {
+    local filename="$(basename "${1}")"
+    IFS=$'\n' local parts=($(echo "${filename}" | tr - "\n"))
+    last_part="${parts[-1]}"
+    pattern="${2}.rpm"
+
+    release="${last_part%.$pattern}"
+    echo "${release}"
+}
+
+# Given a path to an RPM package, echo the package arch.
+#
+# e.g. assets/walrus-5.21-1.noarch.rpm → noarch
+get_rpm_arch() {
+    local filename="$(basename "${1}")"
+    IFS=$'\n' local parts=($(echo "${filename}" | tr . "\n"))
+    echo "${parts[-2]}"
+}
+
+
+# Fetch output_dir from user.
+if [ "$#" -lt 3 ]; then
+    echo 1>&2 'Error: Too few arguments received.'
+    echo 1>&2
+    show_help 1>&2
+    exit 1
+elif [ "$#" -gt 3 ]; then
+    echo 1>&2 'Error: Too many arguments received.'
+    echo 1>&2
+    show_help 1>&2
+    exit 1
+else
+    output_dir="$(realpath --canonicalize-missing "${1}")"
+    assets_dir="$(realpath "${2}")"
```

Comment:
I think this one should use option '--canonicalize-missing'. Unlike del-fixtures-sign.sh,  ${output_dir}" is not expected to exist before running gen-fixtures-delta.sh. It's removed by `make clean`. And gen-fixtures-delta.sh will create this directory ${output_dir}. 
This is also true to other scripts except del-fixture-sign.sh